### PR TITLE
Python script for automatical runtime upgrade on testnets #979 

### PR DESCRIPTION
**Pull request by HaidarJbeily7**

### CODE REVIEW

File: misc/runtime_upgrade/main.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        print(f'Error: {e}')
+        raise e
+    finally:
+        if substrate:
+            substrate.close()
+        
+        
+
+if __name__ == '__main__':
+    main()
```
Comment (Alexey-N-Chernyshov) [inline misc/runtime_upgrade/main.py:93]:
```suggestion
    main()

```

File: misc/runtime_upgrade/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+venv
```
Comment (Alexey-N-Chernyshov) [inline misc/runtime_upgrade/.gitignore:1]:
```suggestion
venv

```

File: misc/runtime_upgrade/main.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    elif args.mnemonic:
+        return Keypair.create_from_mnemonic(args.mnemonic, ss58_format=ss58_format)
+    else:
+        raise Exception("No keypair provided")
+    
+
+def send_extrinsic(substrate_provider: SubstrateInterface, keypair: Keypair, call: GenericCall):
+    extrinsic = substrate_provider.create_signed_extrinsic(call=call, keypair=keypair)
+    try:
+        receipt = substrate_provider.submit_extrinsic(extrinsic, wait_for_inclusion=True)
```
Comment (Alexey-N-Chernyshov) [inline misc/runtime_upgrade/main.py:33]:
```suggestion
        receipt = substrate_provider.submit_extrinsic(extrinsic, wait_for_finalization =True)
```

File: misc/runtime_upgrade/main.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    print('Extrinsic "{}" included in block "{}"'.format(
+        receipt.extrinsic_hash, receipt.block_hash
+    ))
+
+    if receipt.is_success:
+        print('✅ Success, triggered events:')
+        for event in receipt.triggered_events:
+            print(f'* {event.value}')
+    else:
+        print('⚠️ Extrinsic Failed: ', receipt.error_message)
```
Comment (Alexey-N-Chernyshov) [inline misc/runtime_upgrade/main.py:49]: I think it is better to raise an Exception to return 1 from the script.

**Review by Alexey-N-Chernyshov — COMMENTED** [review]
I see the requirement `Should check that after runtime upgrade blocks producing and finalizing`. I think we can raise an exception on extrinsic failure which terminates the script with exit code `1`. Then we can check docker exit code.

**Review by Alexey-N-Chernyshov — APPROVED** [review]
CI complaining on Docker running as root and Python-related code style. Since it is a utility, it is ok for me to merge. If we want to use it in CI/CD, it is better to add non-root user in Docker.
@vovac12, please take a look at this.

**Review by vovac12 — APPROVED** [review]

**Pull request merged**